stdlib/3/unittest/mock.pyi

```diff
@@ -164,35 +164,63 @@ class _patch(Generic[_T]):
     autospec: Any
     kwargs: Mapping[str, Any]
     additional_patchers: Any
-    @overload
-    def __init__(
-        self: _patch[MagicMock],
-        getter: Callable[[], Any],
-        attribute: str,
-        *,
-        spec: Optional[Any],
-        create: bool,
-        spec_set: Optional[Any],
-        autospec: Optional[Any],
-        new_callable: Optional[Any],
-        kwargs: Mapping[str, Any],
-    ) -> None: ...
-    # This overload also covers the case, where new==DEFAULT. In this case, self is _patch[Any].
-    # Ideally we'd be able to add an overload for it so that self is _patch[MagicMock],
-    # but that's impossible with the current type system.
-    @overload
-    def __init__(
-        self: _patch[_T],
-        getter: Callable[[], Any],
-        attribute: str,
-        new: _T,
-        spec: Optional[Any],
-        create: bool,
-        spec_set: Optional[Any],
-        autospec: Optional[Any],
-        new_callable: Optional[Any],
-        kwargs: Mapping[str, Any],
-    ) -> None: ...
+    if sys.version_info >= (3, 8):
+        @overload
+        def __init__(
+            self: _patch[Union[MagicMock, AsyncMock]],
```

Comment:
Not a thorough review, but I would suggest just creating an alias for `Union[MagicMock, AsyncMock]` or just `MagicMock` before this definition. Something like:

```python
if sys.version_info >= (3, 8):
    _Mock = Union[MagicMock, AsyncMock]
else:
    _Mock = MagicMock
```

This way, the size of the branch is much reduced.

Reply:
`_Mock` is a good idea, I'm just not sure if it's worth it. Some IDEs (at least VS Code) will only display the Type Alias instead of the IMO more helpful `Union[MagicMock, AsyncMock]`. Additionally if there is a solution using `overloads`, it will probably need the current syntax.